ui/src/reducers/index.js

```diff
@@ -1,34 +1,64 @@
-import { APPS_FAILURE, APPS_SUCCESS, APPS_REQUEST, APPS_SORT, TOGGLE_HEADER_DROPDOWN } from '../actions/types.js';
+import {
+  APPS_FAILURE,
+  APPS_SUCCESS,
+  APPS_REQUEST,
+  APPS_SORT,
+  TOGGLE_HEADER_DROPDOWN,
+  APP_DETAILS_SORT
+} from '../actions/types.js';
 
 import { SortByDirection, sortable } from '@patternfly/react-table';
 
-const columns = [
-  { title: 'App Name', transforms: [ sortable ] },
-  { title: 'App ID', transforms: [ sortable ] },
-  { title: 'Deployed Versions', transforms: [ sortable ] },
-  { title: 'Current Installs', transforms: [ sortable ] },
-  { title: 'Launches', transforms: [ sortable ] }
-];
-
-const apps = { rows: [], data: {} };
-
-const sortBy = { direction: SortByDirection.asc, index: 0 };
-
 const initialState = {
-  apps: apps,
-  sortBy: sortBy,
-  columns: columns,
+  apps: { rows: [], data: {} },
+  sortBy: { direction: SortByDirection.asc, index: 0 },
+  appDetailsSortDirection: { direction: SortByDirection.asc, index: 0 },
+  columns: [
+    { title: 'App Name', transforms: [ sortable ] },
+    { title: 'App ID', transforms: [ sortable ] },
+    { title: 'Deployed Versions', transforms: [ sortable ] },
+    { title: 'Current Installs', transforms: [ sortable ] },
+    { title: 'Launches', transforms: [ sortable ] }
+  ],
+  appDetailRows: [
+    [ 'v1.0', '25', '321', '2019-01-01 09:55:10', true, 'Deprecated. Please upgrade to latest version' ],
+    [ 'v1.1', '55', '621', '2019-01-11 10:45:03', true, 'Deprecated. Please upgrade to latest version' ],
+    [ 'v1.2', '75', '921', '2019-01-20 12:12:12', false, 'LTS' ],
+    [ 'v1.3', '125', '1221', '2019-01-31 11:05:43', false, 'Curent version' ],
+    [ 'v1.5', '40', '120', '2019-02-15 10:02:50', false, 'Beta version' ]
+  ],
+  appDetailColumns: [
+    { title: 'App Version', transforms: [ sortable ] },
+    { title: 'Current Installs', transforms: [ sortable ] },
+    { title: 'Launches', transforms: [ sortable ] },
+    { title: 'Last Launched', transforms: [ sortable ] },
+    { title: 'Disable on Startup', transforms: [ sortable ] },
+    { title: 'Custom Disable Message', transforms: [ sortable ] }
+  ],
   isAppsRequestFailed: false,
   currentUser: 'currentUser',
   isUserDropdownOpen: false
 };
 
+// returns a new array sorted in preferred order
+const sortRows = (rows, index, order) => {
+  // sort in ascending order
+  const sortedRows = rows.sort((a, b) => (a[index] < b[index] ? -1 : a[index] > b[index] ? 1 : 0));
+
+  // reverse if descending order is preferred
+  if (order !== SortByDirection.asc) {
+    sortedRows.reverse();
+  }
+
+  return sortedRows;
+};
+
 export default (state = initialState, action) => {
   switch (action.type) {
     case APPS_SORT:
+      const order = action.payload.index;
```

Comment:
@damienomurchu I believe this is what's causing the sort to not work. This should be action.payload.direction?

Reply:
I would also suggest using direction for that variable and inside the method that's taking care of the sort as that's the variable used in the state.

Reply:
good spot @laurafitzgerald - fixed and pushed now, can you tell me if issue is still there?

Reply:
The sorting issue is still there on the UI:

![ezgif-1-f954bf450069](https://user-images.githubusercontent.com/11743717/54293089-36a2ba80-45a7-11e9-975e-3db582abe312.gif)

Reply:
both issues above sorted now, so resolving